angular/projects/admin-nrpti/src/app/import/import-csv/import-csv.component.ts

```diff
@@ -0,0 +1,233 @@
+
+      if (this.csvFileErrors && this.csvFileErrors.length) {
+        return;
+      }
+
+      this.csvFileValidated = true;
+    };
+
+    fileReader.readAsText(this.csvFiles[0]);
+  }
+
+  /**
+   * Validate that the csv file contains all required headers.
+   *
+   * @param {string[]} csvRows
+   * @returns
+   * @memberof ImportCSVComponent
+   */
+  validateRequiredHeaders(csvRows: string[]) {
+    if (!csvRows || !csvRows.length) {
+      this.csvFileErrors.push(`Error parsing csv file: ${this.csvFiles[0].name}`);
+      return;
+    }
+
+    // get column header values array
+    let csvHeaderRowArray = String(csvRows[0]).split(',');
+
+    if (!csvHeaderRowArray) {
+      this.csvFileErrors.push(`Error parsing csv file: ${this.csvFiles[0].name}`);
+      return;
+    }
+
+    // convert header values array to lowercase
+    csvHeaderRowArray = csvHeaderRowArray.map(header => header.toLowerCase());
+
+    // get required column headers for the specified data source and record type
+    const requiredHeadersArray = CsvConstants.getCsvRequiredHeadersArray(this.dataSourceType, this.recordType);
+
+    // determine if the csv file is missing any required column headers
+    const missingHeaders = requiredHeadersArray.filter(
+      requiredHeader => !csvHeaderRowArray.includes(requiredHeader.toLowerCase())
+    );
+
+    if (missingHeaders && missingHeaders.length) {
+      this.csvFileErrors.push(`CSV file is missing required column headers: ${missingHeaders}`);
+      return;
+    }
+  }
+
+  /**
+   * Validate that the csv rows contain non-null/non-empty values for all required fields.
+   *
+   * @param {string[]} csvRows
+   * @returns
+   * @memberof ImportCSVComponent
+   */
+  validateRequiredFields(csvRows: string[]) {
+    if (!csvRows || !csvRows.length) {
+      this.csvFileErrors.push(`Error parsing csv file: ${this.csvFiles[0].name}`);
+      return;
+    }
+
+    // get column header values array
+    const csvHeaderRowArray = String(csvRows[0]).split(',');
+
+    // get required fields for the specified data source and record type
+    const requiredFieldsArray = CsvConstants.getCsvRequiredFieldsArray(this.dataSourceType, this.recordType);
+
+    // start loop at index 1, skipping the header row
+    for (let i = 1; i < csvRows.length; i++) {
+      if (!csvRows[i]) {
+        continue;
+      }
+
+      const missingFields = [];
+
+      // get row values array
+      const csvRowValuesArray = String(csvRows[i]).split(',');
+
+      // determine if the csv row is missing any required fields
+      for (const requiredField of requiredFieldsArray) {
+        const requiredFieldIndex = csvHeaderRowArray.indexOf(requiredField);
+
+        if (!csvRowValuesArray[requiredFieldIndex]) {
+          missingFields.push(requiredField);
+        }
+      }
+
+      if (missingFields.length) {
+        this.csvFileErrors.push(`CSV row ${i} is missing required fields: ${missingFields}`);
+      }
+    }
+  }
```

Comment:
We should validate expected formats of things such as DateIssuedFormat and BirthdateFormat here in order to catch it early before allowing the processing button to be enabled.